docs/source/cpp/compute.rst

```diff
@@ -230,10 +234,64 @@ Notes:
   Note that the output can have less than *N* elements if the input has
   less than *N* distinct values.
 
+  The mode kernel is not a proper aggregate (it is actually a vector
+  function, see below).
+
 * \(5) Output is Int64, UInt64 or Float64, depending on the input type.
 
 * \(6) Output is Float64 or input type, depending on QuantileOptions.
 
+  The quantile kernel is not a proper aggregate (it is actually a vector
+  function, see below).
+
+* \(6) tdigest/t-digest computes approximate quantiles, and so only needs a
+  fixed amount of memory. See the `reference implementation
+  <https://github.com/tdunning/t-digest>`_ for details.
+
+Hash Aggregations ("group by")
```

Comment:
I don't know if we want to say "grouped aggregation" rather than "hash aggregation". The former describes the semantics, the latter the implementation. cc @wesm @nealrichardson @ianmcook  for opinions.

Reply:
👍 I like "grouped aggregations"

I think it's also worth explaining briefly what the meaning of "hash" is so that users understand why these function names all begin with `hash_`. I can imagine some confused user thinking this is a list of cryptographic hash functions.